src/devices/vdml_adi.cpp

```diff
@@ -140,4 +157,29 @@ double ADIGyro::get_value() const {
 std::int32_t ADIGyro::reset() const {
 	return adi_gyro_reset(merge_adi_ports(_smart_port, _adi_port));
 }
+
+ADIPotentiometer::ADIPotentiometer(std::uint8_t adi_port, adi_potentiometer_type_e_t potentiometer_type) : ADIAnalogIn(adi_port) { 
+	std::int32_t _port = ext_adi_potentiometer_init(INTERNAL_ADI_PORT, adi_port, potentiometer_type);
+	std::uint8_t _smart_port, _adi_port;
+	_smart_port = get_smart_port();
+	_adi_port = get_adi_port();
+	get_ports(_port, _smart_port, _adi_port);
+	set_smart_port(_smart_port);
+	set_adi_port(_adi_port);
+}
+
+ADIPotentiometer::ADIPotentiometer(ext_adi_port_pair_t port_pair, adi_potentiometer_type_e_t potentiometer_type) : ADIAnalogIn(std::get<1>(port_pair)) { 
+ 	std::int32_t _port = ext_adi_potentiometer_init(port_pair.first, port_pair.second, potentiometer_type);
+	std::uint8_t _smart_port, _adi_port;
+ 	_smart_port = get_smart_port();
+	_adi_port = get_adi_port();
```

Comment:
What's the point of get_smart_port and get_adi_port if the values returned from them are going to be overwritten anyways?